pkg/expr/mathexp/funcs.go

```diff
@@ -113,3 +163,34 @@ func perFloat(e *State, val Value, floatF func(x float64) float64) (Value, error
 
 	return newVal, nil
 }
+
+// perNullableFloat is like perFloat, but takes and returns float pointers instead of floats.
+// This is for instead for functions that need specific null handling.
+// The input float pointer should not be modified in the floatF func.
+func perNullableFloat(e *State, val Value, floatF func(x *float64) *float64) (Value, error) {
```

Comment:
As an alternative (that I implemented in my branch to do the same thing) you can add an argument `nilValue` to the `perFloat` function and control  what nil value is supposed to be.